pinot-common/src/main/java/org/apache/pinot/common/utils/filefetcher/HttpFileFetcher.java

```diff
@@ -16,48 +16,33 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.pinot.common.segment.fetcher;
+package org.apache.pinot.common.utils.filefetcher;
 
 import java.io.File;
 import java.net.URI;
-import java.util.Collections;
-import java.util.Set;
 import org.apache.commons.configuration.Configuration;
 import org.apache.pinot.common.exception.HttpErrorStatusException;
 import org.apache.pinot.common.utils.FileUploadDownloadClient;
-import org.apache.pinot.spi.filesystem.PinotFS;
 import org.apache.pinot.spi.utils.retry.RetryPolicies;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 
-
-public class HttpSegmentFetcher implements SegmentFetcher {
-  protected final Logger _logger = LoggerFactory.getLogger(getClass().getSimpleName());
-
+public class HttpFileFetcher extends BaseFileFetcher {
   protected FileUploadDownloadClient _httpClient;
-  protected int _retryCount;
-  protected int _retryWaitMs;
 
   @Override
-  public void init(Configuration configs) {
-    initHttpClient(configs);
-    _retryCount = configs.getInt(PinotFS.Constants.RETRY, PinotFS.Constants.RETRY_DEFAULT);
-    _retryWaitMs = configs.getInt(PinotFS.Constants.RETRY_WAITIME_MS, PinotFS.Constants.RETRY_WAITIME_MS_DEFAULT);
-  }
-
-  protected void initHttpClient(Configuration configs) {
+  protected void doInit(Configuration config) {
     _httpClient = new FileUploadDownloadClient();
   }
 
   @Override
-  public void fetchSegmentToLocal(final String uri, final File tempFile)
+  public void fetchFileToLocal(URI uri, File dest)
```

Comment:
why not let the caller decide the retry policy (and its parameters, of course). Since we are changing the signatures, maybe we can take RetryPolicy as an argument?

Reply:
Retry policy is configurable via config. In the future if we want to use custom retry, we can expose the `fetchFileToLocalWithoutRetry()` method.